src/Stratis.Bitcoin/Controllers/NodeController.cs

```diff
@@ -354,15 +354,17 @@ public async Task<IActionResult> GetTxOutAsync([FromQuery] string trxid, uint vo
         /// <summary>
         /// Triggers a shutdown of the currently running node.
         /// </summary>
+        /// <param name="delayInSeconds">Delay, in seconds, after which the node will stop.</param>
         /// <returns><see cref="OkResult"/></returns>
         [HttpPost]
         [Route("shutdown")]
         [Route("stop")]
-        public IActionResult Shutdown()
+        public IActionResult Shutdown([FromBody] int delayInSeconds = 0)
         {
-            // Start the node shutdown process.
-            this.fullNode?.Dispose();
-
+            var timer = new System.Timers.Timer(delayInSeconds * 1000);
+            // Start the node shutdown process on timer elapsed.
+            timer.Elapsed += (a, s) => this.fullNode?.Dispose();
+            timer.Start();
```

Comment:
Nice.  Not now, but maybe useful to implement the delay in the `Dispose()` method directly and pass in the `delayInSeconds` (default 0 seconds).

Reply:
I am not sure that I like the idea of a "complex" Dispose() method like that. But we can put some thoughts into it.